odilia/src/tower/mod.rs

```diff
@@ -3,10 +3,13 @@ pub mod choice;
 pub mod from_state;
 pub mod handler;
 pub mod iter_svc;
+pub mod service_ext;
 pub mod service_set;
 pub mod state_svc;
 pub mod sync_try;
+pub mod unwrap_svc;
```

Comment:
what is this for? if I'm thinking correctly here, which I might be not, do we really need a service which unwraps results, isn't this what that service does?

Reply:
The unwrap service provides a way to extract values which are known to be behind the `Infallible` error type. This is useful for us, since our handler functions return a generic type `R`, but `Service` implementers need to return `Result<Response, Error>`, so when we use the `into_service()` conversion function, we make the return type [`Result<R, Infallible>`](https://github.com/odilia-app/odilia/pull/138/files#diff-1457fcdc3d0b4e60ff39a75deef42d4c29c75f1d7989ac031962d7de48267604R76).

After this, however, we need to turn the type `R` into `Result<Vec<Command>, Error>`, which we know it implements from the trait bound `R: TryIntoCommands`. This function needs to be run on `R`, not `Result<R, _>`. This service wrapper provides a convenience combinator to execute this. In previous PRs, I had written `.map_result(|res| res.unwrap().try_into_commands())`, but since we know this is the `Result<_, Infallible>`, I see no reason to write a small wrapper to make this easier and safer—especially if we change the order/add more complex layers later on.